src/schema/sale_artwork.js

```diff
@@ -24,6 +25,51 @@ export const isBiddable = (sale, { artwork: { sold } }) => {
   return !sold && sale.is_auction && sale.auction_state === "open"
 }
 
+const bid_increments_calculator = async ({
+  sale_id,
+  saleLoader,
+  incrementsLoader,
+  minimum_next_bid_cents,
+}) => {
+  return saleLoader(sale_id).then(sale => {
+    if (!sale.increment_strategy) {
+      return Promise.reject("schema/sale_artwork - Missing increment strategy")
+    }
+
+    return incrementsLoader({
+      key: sale.increment_strategy,
+    }).then(incrs => {
+      // We already have the asking price for the lot. Produce a list
+      // of increments beyond that amount. Make a local copy of the
+      // tiers to avoid mutating the cached value.
+      const tiers = incrs[0].increments.slice()
+      const increments = [minimum_next_bid_cents]
+      const limit = BIDDER_POSITION_MAX_BID_AMOUNT_CENTS_LIMIT
+      let current = 0 // Always start from zero, so that all prices are on-increment
+      while (increments.length < 100 && current <= limit) {
+        if (current > minimum_next_bid_cents) increments.push(current)
+        const { to, amount: increase } = tiers[0]
+        current += increase
+        if (current > to && tiers.length > 1) tiers.shift()
+      }
+      return increments
+    })
+  })
+}
```

Comment:
What I meant with `async`, we can replace `then`s with `await`s and flatten out the logic:

```
const bid_increments_calculator = async ({
  sale_id,
  saleLoader,
  incrementsLoader,
  minimum_next_bid_cents,
}) => {
  const sale = await saleLoader(sale_id)
  if (!sale.increment_strategy) {
    return Promise.reject("schema/sale_artwork - Missing increment strategy")
  }

  const incrs = await incrementsLoader({
    key: sale.increment_strategy,
  })

  // We already have the asking price for the lot. Produce a list
  // of increments beyond that amount. Make a local copy of the
  // tiers to avoid mutating the cached value.
  const tiers = incrs[0].increments.slice()
  const increments = [minimum_next_bid_cents]
  const limit = BIDDER_POSITION_MAX_BID_AMOUNT_CENTS_LIMIT
  let current = 0 // Always start from zero, so that all prices are on-increment
  while (increments.length < 100 && current <= limit) {
    if (current > minimum_next_bid_cents) increments.push(current)
    const { to, amount: increase } = tiers[0]
    current += increase
    if (current > to && tiers.length > 1) tiers.shift()
  }
  return increments
}
```

Reply:
Good point, this is done in https://github.com/artsy/metaphysics/pull/1012/commits/741b6aa4a1651465b99583f72c2b2280e3c35ed1